CLI publishing guide for w3 (#1548)
ipfs/ipfs-docs: docs/quickstart/publish_cli.md

---

> title: 'Publish a file with IPFS using a pinning service in a CLI'

Reviewer: ```suggestion
title: 'Publish using the command line'
```

> description: 'Learn how to publish content with IPFS by pinning a file to a pinning service using a CLI.'

Reviewer: ```suggestion
description: 'Learn how to publish content with IPFS by pinning a file to a pinning service using the command line.'
```

---

> # Publish a file with IPFS

Reviewer: ```suggestion
# Publish a file with IPFS using the command line
```

Pinning can be done at various levels, from individual files to entire directories that are addressed by a CID. You can also pin CIDs to multiple IPFS nodes to increase the redundancy and resilience of the file on the network.

## Pinning services

[Pinning services](../concepts/persistence.md#pinning-services) are similar to hosting services, in that they run an IPFS node for you and ensure that your files are available to the IPFS network.

:::callout
Data pinned to the IPFS network is public by default and retrievable by anyone. Avoid publishing private data or adequately encrypt it before publishing.
:::

## Prerequisites

- A unix-like terminal.
- Node.js installed in the terminal environment.
- The [following image](../quickstart/images/welcome-to-IPFS.jpg), downloaded and saved on your computer:

![image](../quickstart/images/welcome-to-IPFS.jpg)

## Install and register to w3

1. Install the [w3 command line interface](https://github.com/web3-storage/w3cli).

>    ```
>    npm install -g @web3-storage/w3cli
>    ```

Reviewer: ```suggestion

   ```shell
   npm install -g @web3-storage/w3cli
   ```
```

1. Authorize the w3 tool to act on behalf of the account associated with your email address `<your@email.com>`:

>    ```
>    w3 authorize <your@email.com>
>    ```

Reviewer: ```suggestion
   ```shell
   w3 authorize <your@email.com>
   ```
```

## Upload and pin a file

1. Create a space for your files.

> 
>    ```shell
>     w3 space create Pictures
> 

Reviewer: ```suggestion

   ```shell
    w3 space create Pictures
   ```
   
```

1. Upload your file.

>    ```

Reviewer: ```suggestion
   ```shell
```

- [Prerequisites](#prerequisites)
- [Uploading and pinning a file](#uploading-and-pinning-a-file)
- [CIDs explained](#cids-explained)
- [Retrieving with a gateway](#retrieving-with-a-gateway)
- [Summary and next steps](#summary-and-next-steps)

## Overview

_Pinning_ refers to the process of ensuring that a particular piece of content is retrievable with IPFS. In other words, pinning is equivalent to storing a file on a computer or server that is connected to the internet, thereby making it available to the rest of the IPFS network.

Pinning can be done at various levels, from individual files to entire directories that are addressed by a CID. You can also pin CIDs to multiple IPFS nodes to increase the redundancy and resilience of the file on the network.

## Pinning services

[Pinning services](../concepts/persistence.md#pinning-services) are similar to hosting services, in that they run an IPFS node for you and ensure that your files are available to the IPFS network.

:::callout
Data pinned to the IPFS network is public by default and retrievable by anyone. Avoid publishing private data or adequately encrypt it before publishing.
:::

## Prerequisites

> - A unix-like terminal
> - Node.js installed to the terminal environment
> - The [following image](../quickstart/images/welcome-to-IPFS.jpg), downloaded and saved on your computer:

Reviewer: ```suggestion
- A unix-like terminal.
- Node.js installed in the terminal environment.
- The [following image](../quickstart/images/welcome-to-IPFS.jpg), downloaded and saved on your computer:
```

![image](../quickstart/images/welcome-to-IPFS.jpg)

## Install and register to w3

> 1. Install the w3 cli tool.

Reviewer: ```suggestion
1. Install the [w3 command line interface](https://github.com/web3-storage/w3cli).
```

   ```
   npm install -g @web3-storage/w3cli
   ```

> 1. Register to w3 using your email.

Reviewer: ```suggestion
1. Authorize the w3 tool to act on behalf of the account associated with your email address `<your@email.com>`:
```

   ```

>    w3 authorize your@email.com

Reviewer: ```suggestion
   w3 authorize <your@email.com>
```

   ```

## Upload and pin a file

1. Create a space for your files, and register it.

   ```
   w3 space create Pictures
   w3 space register

>    ```

Reviewer: Break apart steps for readbability, copyability

For some reason, the change suggester isn't formatting this properly. I'll add it manually

Reviewer: 1. Create a space for your files.

   ```shell
    w3 space create Pictures
   ```
   
1. Register your files.
  
   ```shell
   w3 space register
   ```

Reviewer: added in https://github.com/ipfs/ipfs-docs/pull/1548/commits/3c6a6fb5f71bfc5e41f7609bfd5cf73f1cc22367

1. Upload your file.

   ```
   w3 up welcome-to-IPFS.jpg
   ```

> Congratulations, you have successfully **pinned** a file to IPFS! 🎉

Reviewer: ```suggestion
Congratulations, you have successfully **pinned** a file to IPFS using the w3 cli! 🎉
```

> 1. Create a space for your files, and register it.
> 
>    ```
>    w3 space create Pictures
>    w3 space register
>    ```

Reviewer: ```suggestion
1. Create a space for your files.

   ```shell
    w3 space create Pictures
   ```
   
1. Register your files.
  
   ```shell
   w3 space register
   ```
```
